### PR TITLE
Merge : Serialize output mask; Enable user arguments for callable mode

### PR DESCRIPTION
**Pull request by farizrahman4u**

### CODE REVIEW

File: keras/engine/topology.py [hunk 1]
```diff
@@ -1239,9 +1242,10 @@ def call(self, inputs, mask=None):
                             '(at least 2). Got: ' + str(inputs))
         # Case: "mode" is a lambda or function.
         if hasattr(self.mode, '__call__'):
-            # TODO: consider making it possible to
-            # pass custom arguments to lambda.
-            arguments = {}
+            arguments = self.arguments
+            arg_spec = inspect.getargspec(self.mode)
+            if 'mask' in arg_spec.args:
+                    arguments['mask'] = mask
```
Comment (fchollet) [inline keras/engine/topology.py]: Bad indentation.

**Comment by farizrahman4u** [comment]
What are the possible types of output_mask? Anything other than None, Function, Lambda ?

**Comment by farizrahman4u** [comment]
Tests passing on theano backend.. failing on tensorflow.

**Comment by farizrahman4u** [comment]
@fchollet I don't have tensorflow; help!

File: tests/keras/layers/test_core.py [hunk 2]
```diff
@@ -53,7 +53,8 @@ def test_merge():
     input_b = Input(shape=input_shapes[1][1:])
     merged = merge([input_a, input_b],
                    mode=lambda tup: K.concatenate([tup[0], tup[1]]),
-                   output_shape=lambda tup: (tup[0][:-1],) + (tup[0][-1] + tup[1][-1],))
+                   output_shape=lambda tup: tup[0][:-1] + (tup[0][-1] + tup[1][-1],))
+    model = Model([input_a, input_b], merged)
```
Comment (farizrahman4u) [inline tests/keras/layers/test_core.py]:
```python
model = Model([input_a, input_b], merged)
```
@fchollet  Lol.. I am adding this line now. Which mean you weren't really testing anything here.

**Comment by farizrahman4u** [comment]
@fchollet Done.

**Comment by farizrahman4u** [comment]
ping @fchollet

**Comment by fchollet** [comment]
LGTM, thanks.
